File: clients/go/examples/collections-find-one-and-replace-upsert.go
package main

import (
	"context"
	"log"

	"github.com/datastax/astra-db-go/v2/astra"
	"github.com/datastax/astra-db-go/v2/astra/filter"
	"github.com/datastax/astra-db-go/v2/astra/options"
)

func main() {
	ctx := context.Background()

	// Get an existing collection
	client := astra.NewClient()

	database := client.Database(
		"**API_ENDPOINT**",
		options.API().SetToken("**APPLICATION_TOKEN**"),
	)

	collection := database.Collection("**COLLECTION_NAME**")

	// Replace a document
	err := collection.FindOneAndReplace(
		ctx,
		filter.And(
			filter.Eq("is_checked_out", false),
			filter.Lt("number_of_pages", 300),
		),
		map[string]any{"is_checked_out": true, "borrower": "Brook Reed"},
		options.CollectionFindOneAndReplace().SetUpsert(true),
	).Err()
	if err != nil {
		log.Print(err)
	}
}

File: clients/typescript/examples/collections-find-one-and-replace-upsert.ts
import { DataAPIClient } from "@datastax/astra-db-ts";

// Get an existing collection
const client = new DataAPIClient();
const database = client.db("**API_ENDPOINT**", {
  token: "**APPLICATION_TOKEN**",
});
const collection = database.collection("**COLLECTION_NAME**");

// Replace a document
(async function () {
  await collection.findOneAndReplace(
    {
      $and: [{ is_checked_out: false }, { number_of_pages: { $lt: 300 } }],
    },
    {
      is_checked_out: true,
      borrower: "Brook Reed",
    },
    { upsert: true },
  );
})();
